### ISSUE TITLE
Differentiate CO-OPS Stations with Just Tide Predictions vs Ones with Water Level Measurements

### ISSUE THREAD

**Issue by jamesdoyle21** [asascience/ott-coops#434]
OceansMap Bug Tracker line 40

This is an issue that has layers and is causing problems down the line. The problem is that CO-OPS stations labeled "Water Levels" contain 2 types of station: stations with both measured water level data and tidal predictions and stations with only tidal predictions. We need to differentiate these because the stations with tidal predictions only have no real-time data - just predictions.

Matt, Chris and I did not have a clear solution to this. Ideally, we would add a 5th category to the CO-OPS stations titled "Tide Predictions" which contains all the stations with only tide predictions and nothing else. This is a challenging problem though apparently, so it may require some discussion of how to do this here.

This issue directly relates to #435. If we can create a separate group for the tide prediction only stations, we can remove these from the validation tool completely to solve #435.

![image](https://user-images.githubusercontent.com/82955395/163227611-729bcd03-435a-4833-9ba0-0844335a9b70.png)

**Comment by kvmartin001** [comment]
For https://github.com/asascience/ott-coops/issues/429 I'm adding the category as an actual data attribute, so the category will be readable by the validation tool.

**Comment by jamesdoyle21** [comment]
The red Water Levels stations are coming up orange on my map. This might be because all the statuions are out of date, but either way, this should be easier to read.

Also, I am not seeing this change in the Client Demo instance

Putting this back into In Progress @kvmartin001 

![image](https://user-images.githubusercontent.com/82955395/163842224-0936b90e-0a80-47db-bcf0-a22f2273fbd3.png)

**Comment by jamesdoyle21** [comment]
The other thing is there are many stations with Water Level listed as a parameter, but never have data for this.

@cpaternostro can you confirm, should data stations like the one listed below be labeled as a Water Level Station or a Tide Prediction station?

![image](https://user-images.githubusercontent.com/82955395/163842758-6b56fb9b-3f9d-4fdd-9f39-e990619bc527.png)

**Comment by kvmartin001** [comment]
@jamesdoyle21 Yeah the color problem is because of the out-of-date fade. Didn't notice on Fri because most stations were current. How's this?
![image](https://user-images.githubusercontent.com/56608780/163850578-d156ecb8-ae9c-41ce-973c-ce3dc75339b0.png)

**Comment by jamesdoyle21** [comment]
Much better

**Comment by cpaternostro** [comment]
The data inventory looks like station 8633532 Tangier Island, VA has been
removed.  So, YES, it is solely a Tide prediction station.
https://tidesandcurrents.noaa.gov/inventory.html?id=8633532
But I'm checking others like station 8630111 JESTERS ISLAND, CHINCOTEAGUE
BAY, VA, which is marked as a water levels station, but it is hasn't had
data since 1985 so it should be marked as a Tide predictions station.

What is the logic in painting a station yellow (prediction) or red dot
(active WL)?  How can I help correct this?

--

Chris Paternostro

https://tidesandcurrents.noaa.gov/

**Comment by jamesdoyle21** [comment]
@kvmartin001 see Chris's comment here. Many of the stations we have labeled red (Active WL) have not had water level data here in 30+ years. Correct me if I am wrong here Kevin, but I believe any station that has ever had water level data at one point is labeled red in OceansMap right now. So, @cpaternostro, we would need a list of the co-ops stations that are just tide prediction stations so Kevin can go in and set each one. Let me know if there is an easier way around this.

**Comment by jamesdoyle21** [comment]
This is also a backend issue because parameters that haven't had data in 37 years probably shouldn't be showing up in OceansMap..

**Comment by kvmartin001** [comment]
@jamesdoyle21 yes, you're right about the labels. I'm labeling stations by the Parameters list, which is in the OceansMap Catalog Item. The parameters come directly from the data service, https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ in this case. Maybe our backend needs a filter that removes parameters from the list if they aren't considered current.

As you said, I could also add logic on the frontend that would paint stations based on their ID.

**Comment by mpiannucci** [comment]
I would really rather not do that @kvmartin001 It should be all driven from the data availabel in the catalog

**Comment by cpaternostro** [comment]
All historic water level stations should have prediction data available.
I guess what the catalog is trying to do is know which stations have
available data.  Could the list on the left of that Opendap SOS service be
the reference of stations that have present day data?  Paint the stations
in that Stations listings for Water Level red and the rest yellow.
STATIONS LISTINGS

   - Active Currents
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=4>
   - Survey Currents
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=5>
   - *Water Level*
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=6>
   - Water Level One Minute
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=7>
   - Meteorological & Ancillary
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=8>
   - Conductivity
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=9>
   - Rain Fall
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=10>
   - Relative Humidity
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=11>
   - Visibility
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=12>
   - Harmonic Constituents
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=13>
   - Datums
   <https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=14>



--

Chris Paternostro

https://tidesandcurrents.noaa.gov/

**Comment by jamesdoyle21** [comment]
I am going to leave this issue open. Let's plan to discuss with @cpaternostro on Tuesday

**Comment by mpiannucci** [comment]
Data catalog should be updated to use get caps to query station availabaility

https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/SOS?service=SOS&request=GetCapabilities&version=1.0.0

**Comment by jamesdoyle21** [comment]
@cpaternostro we now have Tide Prediction stations separated from Water Level stations. Can you take a look at this and let us know your thoughts? Most of the stations labeled Water Level only have tide prediction data because the water level data gives a warning and is grayed out. Curious on your thoughts how to best handle these labels

**Comment by kvmartin001** [comment]
We still need to edit Data Catalog to add support for Water Leve active / inactive status.

@mpiannucci On second look here, I'm not seeing any stations tagged as active / inactive in the GetCaps (link above).

There's a DescribeSensor request that looks promising. You can specify a SOS procedure, in this case `urn:ioos:network:NOAA.NOS.CO-OPS:WaterLevelActive`, and it gives you the list of active water level stations:
https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/SOS?service=SOS&request=DescribeSensor&version=1.0.0&outputFormat=text%2Fxml%3Bsubtype%3D%22sensorML%2F1.0.1%2Fprofiles%2Fioos_sos%2F1.0%22&procedure=urn:ioos:network:NOAA.NOS.CO-OPS:WaterLevelActive

About the DescribeSensor request
https://opendap.co-ops.nos.noaa.gov/ioos-dif-sos/ClientGetter?p=3

**Comment by mpiannucci** [comment]
Are we using opendap or scidap currently for coops data? We should have a very real discussion about whether we keep using scidap for this or hit their opendap directly. For coops OTT, I am partial to hitting them directly. The other versions can use scidap. @kvmartin001  can you do a look through how we do the harvesting and datacatalot stuff currently and see if this could be added to how we do things now?

**Comment by mpiannucci** [comment]
See our harvester here 

https://github.com/asascience/data-harvesters/blob/master/data_harvesters/coops_metadata.json
https://github.com/asascience/data-harvesters/blob/master/data_harvesters/harvest_coops.py
https://github.com/asascience/data-harvesters/blob/master/data_harvesters/harvest_coops_new.py

And the data_catalog reader: 

https://github.com/asascience/data-catalog/blob/master/data_catalog/maracoos/coops.py

**Comment by kvmartin001** [comment]
@mpiannucci Yeah I'll take a look. We're currently using SciDAP for CO-OPS simply because our function always tries SciDAP first.
Where should the info be stored in the catalog? Maybe we add a "WaterLevelActive" string to the `tags` list?
